src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metric_flyout/dimension_badges.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { EuiBadge, EuiToken, EuiText, EuiToolTip } from '@elastic/eui';
+import React from 'react';
+
+interface Dimension {
+  name: string;
+  type: string;
+  description?: string;
+}
+
+interface DimensionBadgesProps {
+  dimensions: Dimension[];
+  metricName: string;
+  maxDisplay?: number;
+}
+
+export const DimensionBadges = ({
+  dimensions,
+  metricName,
+  maxDisplay = 10,
+}: DimensionBadgesProps) => {
+  // Extract top-level namespace from metric name (e.g., "system" from "system.network.in.bytes")
+  const topLevelNamespace = metricName.split('.')[0] + '.';
+
+  // Sort dimensions to prioritize attributes.* and top-level namespace
+  const sortedDimensions = [...dimensions].sort((a, b) => {
+    const aIsAttributes = a.name.startsWith('attributes.');
+    const bIsAttributes = b.name.startsWith('attributes.');
+    const aIsTopLevel = a.name.startsWith(topLevelNamespace);
+    const bIsTopLevel = b.name.startsWith(topLevelNamespace);
+
+    // Priority: attributes.* first, then top-level namespace, then others
+    if (aIsAttributes && !bIsAttributes) return -1;
+    if (!aIsAttributes && bIsAttributes) return 1;
+    if (aIsTopLevel && !bIsTopLevel && !aIsAttributes && !bIsAttributes) return -1;
+    if (!aIsTopLevel && bIsTopLevel && !aIsAttributes && !bIsAttributes) return 1;
+
+    // Alphabetical for same priority
+    return a.name.localeCompare(b.name);
+  });
```

Comment:
Maybe we could move this to `dimension_utils`, and possibly simplify it?

Would sth like this work?

```ts
 const getSortPriority = (name: string): number => {
  if (name.startsWith('attributes.')) {
    return 0;  
  }
  if (name.startsWith(`${topLevelNamespace}.`)) {
      return 1;
  }
  return 2;
};

export const sortDimensions = (
  dimensions: Array<{ name: string; type: string }>,
  metricName: string
) => {
  const topLevelNamespace = getTopLevelNamespace(metricName);

  return [...dimensions].sort((a, b) => {
    const priorityDiff = getPriority(a.name) - getPriority(b.name);
    return priorityDiff !== 0 ? priorityDiff : a.name.localeCompare(b.name);
  });
};

It'd be nice to have tests for this function too.